utils.py

```diff
@@ -29,3 +32,65 @@ def deserialize_from_dill(dill_path: Any) -> Any:
     logging.error('Failed to deserialize %s: File does not exist: %s',
                   dill_path, e)
     return None
+
+
+def retryable(exceptions=None,
+              default_retries=5,
+              delay_fn=None,
+              other_exceptions=None,
+              default_return=None):
+  """
+    Decorator that retries the function on specified exceptions.
+    :param exceptions: List/Set of exceptions or a dictionary of exceptions with
+      custom retry counts.
+    :param default_retries: Number of retries if no custom count is provided.
+    :param delay_fn: Function to determine the delay between retries. Default
+      is random between 0-60 seconds.
+    :param default_return: The value returned if all retries fail.
+    """
+  exception_config = {
+      exc: default_retries for exc in exceptions or {}
```

Comment:
Do we need the `dict` support here to support different retry configs per exception? Would be good to follow YAGNI here and keep things simple until we actually need it.

Reply:
I added this mainly for errors related to exceeding the quota limit.
We can retry them a large number of times (e.g., 100), while only retry others (e.g., the VertexAI internal API error we saw earlier) for max 5 times.